src/simtools/ray_tracing/incident_angles.py

```diff
@@ -0,0 +1,735 @@
+            Output arrays to append into.
+        """
+        sx_ok, sx_cm = self._parse_float(parts, col_idx.get("sec_x"))
+        sy_ok, sy_cm = self._parse_float(parts, col_idx.get("sec_y"))
+        if sx_ok and sy_ok:
+            r2_m = ((sx_cm**2 + sy_cm**2) ** 0.5) / 100.0
+            if secondary_radius_m is not None:
+                secondary_radius_m.append(r2_m)
+            if secondary_hit_x_m is not None:
+                secondary_hit_x_m.append(sx_cm / 100.0)
+            if secondary_hit_y_m is not None:
+                secondary_hit_y_m.append(sy_cm / 100.0)
+            return
+        if secondary_radius_m is not None:
+            secondary_radius_m.append(float("nan"))
+        if secondary_hit_x_m is not None:
+            secondary_hit_x_m.append(float("nan"))
+        if secondary_hit_y_m is not None:
+            secondary_hit_y_m.append(float("nan"))
+
+    @staticmethod
+    def _match_header_column(col_pat, raw):
+        """Parse a header line for a known angle column.
+
+        Parameters
+        ----------
+        col_pat : Pattern[str]
+            Compiled regular expression matching ``# Column N`` prefix.
+        raw : str
+            Raw header line.
+
+        Returns
+        -------
+        tuple[str, int] | None
+            ``(kind, column_number)`` when recognized, otherwise ``None``.
+        """
+        s = raw.strip()
+        if s and ":" in s:
+            prefix, desc = s.split(":", 1)
+            m = col_pat.match(prefix)
+            if m:
+                num = int(m.group(1))
+                desc = desc.strip().lower()
+                if "angle of incidence at focal surface" in desc and "optical axis" in desc:
+                    return "focal", num
+                if re.search(r"angle of incidence\s+on(to)?\s+primary mirror", desc):
+                    return "primary", num
+                if re.search(r"angle of incidence\s+on(to)?\s+secondary mirror", desc):
+                    return "secondary", num
+        return None
+
+    def _save_results(self):
+        """Save the results to an ECSV file in the results directory if available."""
+        if self.results is None or len(self.results) == 0:
+            self.logger.warning("No results to save")
+            return
+        output_file = self.results_dir / f"incident_angles_{self._label_suffix()}.ecsv"
+        self.results.write(output_file, format="ascii.ecsv", overwrite=True)
+
+    def export_results(self):
+        """Export the results ECSV and a short text summary file."""
+        if self.results is None or len(self.results) == 0:
+            self.logger.error("Cannot export results because they do not exist")
+            return
+        table_file = self.results_dir / f"incident_angles_{self._label_suffix()}.ecsv"
+        self.results.write(table_file, format="ascii.ecsv", overwrite=True)
+        summary_file = self.results_dir / f"incident_angles_summary_{self._label_suffix()}.txt"
```

Comment:
Trying to understand better the summary file. Is this metadata? Those values are from the tool configuration? Then they should be saved as metadata. 

Otherwise it would be better to save this not as txt, but as yaml or json to guarantee machine readability.